www/src/components/CustomThemesForm.tsx

```diff
@@ -0,0 +1,190 @@
+import React, {
+  useRef,
+  useEffect,
+  useImperativeHandle,
+  forwardRef,
+} from 'react';
+import { useForm, useFieldArray, Controller } from 'react-hook-form';
+import {
+  Form,
+  Button,
+  IconButton,
+  Stack,
+} from '~paragon-react';
+import { Close, Plus } from '~paragon-icons';
+
+interface CustomThemesFormProps {
+  initialTheme: {
+    name: string;
+    urls: string[];
+  } | null;
+  onSave: (theme: { name: string; urls: string[] }) => void;
+}
+
+export interface CustomThemesFormRef {
+  submitForm: () => void;
+}
+
+interface FormData {
+  name: string;
+  urls: { url: string }[];
+}
+
+function isValidCssUrl(url: string) {
+  try {
+    const u = new URL(url);
+    return u.protocol.startsWith('http') && u.pathname.endsWith('.css');
+  } catch {
+    return false;
+  }
+}
+
+const CustomThemesForm = forwardRef<CustomThemesFormRef, CustomThemesFormProps>(({ initialTheme, onSave }, ref) => {
+  const {
+    register,
+    handleSubmit,
+    control,
+    formState: { errors },
+    trigger,
+    getValues,
+  } = useForm<FormData>({
+    defaultValues: {
+      name: initialTheme?.name || '',
+      urls: initialTheme?.urls.length ? initialTheme.urls.map(url => ({ url })) : [{ url: '' }],
+    },
+    mode: 'onTouched',
+  });
+
+  const { fields, append, remove } = useFieldArray({
+    control,
+    name: 'urls',
+  });
+
+  const urlInputRefs = useRef<(HTMLInputElement | null)[]>([]);
+  const prevUrlsLength = useRef(fields.length);
+
+  useEffect(() => {
+    if (fields.length > prevUrlsLength.current) {
+      urlInputRefs.current[fields.length - 1]?.focus();
+    }
+    prevUrlsLength.current = fields.length;
+  }, [fields.length]);
+
+  const onSubmit = (data: FormData) => {
+    onSave({
+      name: data.name.trim(),
+      urls: data.urls.map(u => u.url.trim()).filter(url => url.length > 0),
+    });
+  };
+
+  const submitForm = async () => {
+    const isValid = await trigger();
+    if (isValid) {
+      const data = getValues();
+      onSubmit(data);
+    }
+  };
+
+  useImperativeHandle(ref, () => ({
+    submitForm,
+  }));
+
+  const handleAddUrl = () => {
+    append({ url: '' });
+  };
```

Comment:
It seems like something odd is happening when adding URLs without filling in fields


https://github.com/user-attachments/assets/6c3b03a3-3d82-45bb-b5d0-bbb3d6fc9925

Reply:
This behavior is technically "as designed" in the current state. When you click "Add another URL", it moves focus into the newly rendered input. Then, when you click "Add another URL" again, the input field has been "touched" (blurred) so `react-hook-form` runs its validation to show the "CSS URL is required" message.

I think the UX improvement here would be to prevent clicking "Add another URL" (disabled button) if you there's have an empty input displayed and available to use.

Reply:
Updated in latest commits!